jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionConfigDialog.kt

```diff
@@ -0,0 +1,112 @@
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.util.ExceptionUtil
+import org.jetbrains.annotations.TestOnly
+import software.amazon.awssdk.services.lambda.LambdaClient
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.help.HelpIds
+import software.aws.toolkits.jetbrains.services.lambda.LambdaFunction
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamOptions
+import software.aws.toolkits.jetbrains.utils.lambdaTracingConfigIsAvailable
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.LambdaTelemetry
+import software.aws.toolkits.telemetry.Result
+import javax.swing.Action
+import javax.swing.JComponent
+
+private val NOTIFICATION_TITLE = message("lambda.service_name")
+
+class UpdateFunctionConfigDialog(private val project: Project, private val initialSettings: LambdaFunction) : DialogWrapper(project) {
+    private val view = UpdateFunctionConfigPanel(project)
+
+    private val action: OkAction = object : OkAction() {
+        init {
+            putValue(Action.NAME, message("lambda.upload.update_settings_button.title"))
+        }
+    }
+
+    init {
+        super.init()
+        title = message("lambda.upload.updateConfiguration.title", initialSettings.name)
+
+        view.name.text = initialSettings.name
+        view.description.text = initialSettings.description
+
+        with(view.configSettings) {
+            runtime.selectedItem = initialSettings.runtime
+            handlerPanel.setRuntime(initialSettings.runtime)
+            handlerPanel.handler.text = initialSettings.handler
+            envVars.envVars = initialSettings.envVariables ?: emptyMap()
+            timeoutSlider.value = initialSettings.timeout
+            memorySlider.value = initialSettings.memorySize
+            iamRole.selectedItem = initialSettings.role
+            xrayEnabled.isSelected = initialSettings.xrayEnabled
+
+            setXrayControlVisibility(lambdaTracingConfigIsAvailable(AwsConnectionManager.getInstance(project).activeRegion))
+        }
+    }
+
+    override fun createCenterPanel(): JComponent? = view.content
+
+    override fun getPreferredFocusedComponent(): JComponent? = view.configSettings.handlerPanel.handler
+
+    override fun doValidate(): ValidationInfo? = view.validatePanel()
+
+    override fun getOKAction(): Action = action
+
+    override fun doCancelAction() {
+        LambdaTelemetry.editFunction(project, result = Result.Cancelled)
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        val functionDetails = viewToFunctionDetails()
+        val lambdaClient: LambdaClient = project.awsClient()
```

Comment:
nit: move this into executeOnPooledThread so if it fails it handles it

Reply:
It won't handle as it is written right now.

Reply:
The try catch is on the Completion stages and this is used before that point